homeassistant/components/switch/konnected.py

```diff
@@ -79,8 +78,11 @@ def _set_state(self, state):
         _LOGGER.debug('Setting status of %s actuator pin %s to %s',
                       self._device_id, self.name, state)
 
-    @asyncio.coroutine
-    def async_set_state(self, state):
+    async def async_added_to_hass(self):
+        """Register update callback."""
+        self._data['entity_id'] = self.entity_id
+
```

Comment:
The switch platform doesn't connect to the dispatcher?

Reply:
Right. The switch platform is only making HTTP calls out to the device to tell it to turn on or off. In cdca435 I just updated these methods to update the state based on the feedback from the device as you suggested. Please take a look.